File: 942.增减字符串匹配.cpp
/*
 * @lc app=leetcode.cn id=942 lang=cpp
 *
 * [942] 增减字符串匹配
 */

// @lc code=start
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int n = s.length();
        vector<int> ans(n+1);
        int l = 0;
        int r = n;

        for( int i=0; i<n; i++ ) {
            char c = s[i];
            if( c == 'I' ) {
                ans[i] = l;;
                l++;
            } else {
                ans[i] = r;
                r--;
            }
        }

        ans[n] = l;

        return ans;
    }
};
// @lc code=end

File: 942.增减字符串匹配.rs
/*
 * @lc app=leetcode.cn id=942 lang=rust
 *
 * [942] 增减字符串匹配
 */

// @lc code=start
impl Solution {
    pub fn di_string_match(s: String) -> Vec<i32> {
        let mut ans = vec![0; s.len() + 1];
        let (mut l, mut r) = (0, s.len() as i32);

        s.chars().enumerate().for_each(|(i, b)| if b == 'I' {
            ans[i] = l;
            l += 1
        } else {
            ans[i] = r;
            r -= 1
        });
        ans[s.len()] = l;
        ans
    }
}
// @lc code=end

